File: typescript_api_startup_sl8s/app/src/model_wrapper_repository/example_wrapper_repository/example_wrapper_repository.ts
import { BaseModelWrapperRepository, EnumGuilty, LocalException, Result, ResultWithModelWrapper, ResultWithListModelsWrapper } from "@antonpichka/vitruvius";
import ExampleWrapper from "../../model/example/example_wrapper/example_wrapper";
import ListExampleWrapper from "../../model/example/example_wrapper/list_example_wrapper";

class ExampleWrapperRepository extends BaseModelWrapperRepository {
    public constructor() {
        super();
    }

    public override dispose(): void {
    }
    
    public async getExampleParameterUnknown(): Promise<ResultWithModelWrapper> {
        throw Error();
    }
}

export default ExampleWrapperRepository;

File: dart_web_startup_sl8s/app/lib/model_wrapper_repository/example_wrapper_repository/example_wrapper_repository.dart
import 'package:dart_web_startup_sl8s/model/example/example_wrapper/example_wrapper.dart';
import 'package:dart_web_startup_sl8s/model/example/example_wrapper/list_example_wrapper.dart';
import 'package:library_architecture_mvvm_modify/library_architecture_mvvm_modify.dart';
import 'package:meta/meta.dart';

@immutable
base class ExampleWrapperRepository<T extends ExampleWrapper, Y extends ListExampleWrapper> extends BaseModelWrapperRepository {
  const ExampleWrapperRepository();

  @override
  void dispose() {
  }

  Future<ResultWithModelWrapper<T>> getExampleParameterUnknown() async {
    throw Exception();
  }
}